app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,36 @@
+class FavoritesController < ApplicationController
+  def index
+    current_profile = Profile.find(params[:favorite_profile_id])
+    @collections = current_profile.favorites
+  end
+
+  def create
+    profile          = Profile.find(params[:favorite_profile_id])
+    favorite_profile = Profile.find_by(user_id: current_user.id)
+    collection       = Favorite.new(profile: profile, favorite_profile: favorite_profile)
```

Comment:
It's a bit confusing naming for only one record. Collection should includes a coupe of objects. I'd prefer to name it `new_favorite`

Reply:
I'll renamed it to just object in an agreement with ApplicationController